### PR TITLE
fix bug - guard could not guess guard

### PR DESCRIPTION
**Pull request by FuyaosGithub**
fix bug - guard could not guess guard
#37

### CODE REVIEW

**Review by qrtt1 — CHANGES_REQUESTED** [review]
需要有 testcast 來驗證它。

可以補在 `guard.feature`

File: love_letter/models/cards.py [hunk 1]
```diff
@@ -57,6 +57,8 @@ class GuardCard(Card):
     quantity = 5
 
     def trigger_effect(self, card_holder: "Player", chosen_player: "Player" = None, with_card: "Card" = None):
+        if with_card == GuardCard:
```
Comment (qrtt1) [inline love_letter/models/cards.py:60]: object instance 與 class 相比不會相等，得使用 `isinstance`

https://docs.python.org/3/library/functions.html#isinstance

**Review by qrtt1 — CHANGES_REQUESTED** [review]

File: love_letter/models/cards.py (hunk 1 above)
Comment (IamSaker) [reply love_letter/models/cards.py:60]: 再麻煩 @FuyaosGithub 調整
```python
if with_card == GuardCard isinstance(:
```

Comment (FuyaosGithub) [reply love_letter/models/cards.py:60]: okok

Comment (FuyaosGithub) [reply love_letter/models/cards.py]: > 再麻煩 @FuyaosGithub 調整
> 
> ```python
> if isinstance(with_card, GuardCard):
>     return False
> ```

此處已修正

**Review by IamSaker — APPROVED** [review]

**Review by qrtt1 — APPROVED** [review]

**Pull request merged**